CRM/Core/BAO/Block.php

```diff
@@ -131,20 +131,15 @@ public static function dataExists($blockFields, &$params) {
    * Check if the current block exits.
    *
    * @param string $blockName
-   *   Bloack name.
+   *   Block name.
    * @param array $params
-   *   Associated array of submitted fields.
+   *   Array of submitted fields.
    *
    * @return bool
-   *   true if the block exits, otherwise false
+   *   true if the block is in the params and is an array
    */
-  public static function blockExists($blockName, &$params) {
-    // return if no data present
-    if (empty($params[$blockName]) || !is_array($params[$blockName])) {
-      return FALSE;
-    }
-
-    return TRUE;
+  public static function blockExists($blockName, $params) {
+    return is_array($params[$blockName] ?? NULL);
```

Comment:
That's not quite equivalent to the above statement. Maybe this doesn't matter in practice, but before it was returning FALSE when `$blockName === []` and now it would return TRUE in that circumstance. To fix:
```suggestion
    return !empty($params[$blockName]) && is_array($params[$blockName]);
```

Reply:
OK - I've switched to your line...